File: solutions/3208. Alternating Groups II/3208.java
class Solution {
  public int numberOfAlternatingGroups(int[] colors, int k) {
    final int n = colors.length;
    int ans = 0;
    int alternating = 1;

    for (int i = 0; i < n + k - 2; ++i) {
      alternating = colors[i % n] == colors[(i - 1 + n) % n] ? 1 : alternating + 1;
      if (alternating >= k)
        ++ans;
    }

    return ans;
  }
}

File: solutions/3208. Alternating Groups II/3208.py
class Solution:
  def numberOfAlternatingGroups(self, colors: list[int], k: int) -> int:
    n = len(colors)
    ans = 0
    alternating = 1

    for i in range(n + k - 2):
      alternating = (1 if colors[i % n] == colors[(i - 1) % n]
                     else alternating + 1)
      if alternating >= k:
        ans += 1

    return ans
